components/remote_settings/src/storage.rs

```diff
@@ -27,27 +27,51 @@ use sql_support::{open_database::open_database_with_flags, ConnExt};
 /// passes a new server or bucket to `update_config`, we don't want to be using cached data from
 /// the previous config.
 pub struct Storage {
-    conn: Connection,
+    path: Utf8PathBuf,
+    conn: ConnectionCell,
 }
 
 impl Storage {
     pub fn new(path: Utf8PathBuf) -> Result<Self> {
-        let conn = open_database_with_flags(
+        Ok(Self {
             path,
-            OpenFlags::default(),
-            &RemoteSettingsConnectionInitializer,
-        )?;
-        Ok(Self { conn })
+            conn: ConnectionCell::Uninitialized,
+        })
+    }
+
+    fn transaction(&mut self) -> Result<Transaction<'_>> {
+        match &self.conn {
+            ConnectionCell::Uninitialized => {
+                if self.path != ":memory:" && std::fs::exists(&self.path)? {
+                    std::fs::create_dir(&self.path)?;
```

Comment:
@mhammond I think there's some more motivation for this `create_dir` call now.  The issue I'm trying to work around is that creating a directory in JS requires an async call and we're currently creating `RemoteSettingsService` outside an async context.  I think it would be possible to update the JS code to do that, but it would mean wrapping `RemoteSettingsService` in some sort of async/lazy wrapper -- which would then mean all the suggest/relevancy components also need this wrapping.